### DISCUSSION TITLE
dynamic key in application properties

### DISCUSSION THREAD

**Discussion by gituharish** [quarkiverse/quarkus-cxf#474] [Q&A]
Is it possible to use dynamic configurable keys in application properties? 
For example, quarkus.cxf.endpoint. ${ENDPOINT_URL} .implementor = $${ENDPOINTIMPL} instead of quarkus.cxf.endpoint."/fruit1".implementor=org.acme.cxf.Fruit1WebServiceImpl
quarkus.cxf.endpoint."/fruit2".implementor=org.acme.cxf.Fruit2WebServiceImpl
quarkus.cxf.endpoint."/fruit3".implementor=org.acme.cxf.Fruit3WebServiceImpl
quarkus.cxf.endpoint."/fruit4".implementor=org.acme.cxf.Fruit4WebServiceImpl

The intention is to use the same codebase for deploying multiple services in the pipeline but every service should expose only one  of the endpoints out of the four.

**Comment by shumonsharif** [comment]
Hi @gituharish I believe the closest thing to achieving your objective may be to look into Quarkus [profiles](https://quarkus.io/guides/config-reference#profiles).

We had a similar discussion in the past regarding dynamic keys, which we had decided not to pursue at that point in time.
https://github.com/quarkiverse/quarkus-cxf/issues/131

Keep me posted please if either of the above options works for you.  If not, I can look further into it.
